### ISSUE TITLE
Improve semantic of TransposeConv

### ISSUE THREAD

**Issue by jackwish** [jackwish/tflite2onnx#26]
Serve as a list of trivial items that are not tracked by other issues.

- [x] [Merge all trivial activation into one operator converter class](https://github.com/jackwish/tflite2onnx/pull/25#discussion_r521355235)
- [x] [Adjust `createVector()` such that the data type of the created tensor is parsed from the NDArray rather than the `ref` tensor.](https://github.com/jackwish/tflite2onnx/pull/25#discussion_r521374491)
- [x] [Have a util function to create _empty_ tensor for optional inputs of ONNX operator](https://github.com/jackwish/tflite2onnx/pull/25/files/01e1c8034f80b31130a296d5827087603ca86722#diff-07134a3031c180fdb9dc347a0d69837c0909fefddd733199210e3d34ae56db0fR87)
- [x] [`assert` with `TypeMapping` to check TFLite opcode](https://github.com/jackwish/tflite2onnx/pull/25/files?file-filters%5B%5D=.py&file-filters%5B%5D=.tflite#diff-561e27478cf0dba165cee501175573b3b39d33e4b2022018024a9b148cc5e5e7R41)
 - [x] [Minor comment update for `Resize`](https://github.com/jackwish/tflite2onnx/pull/25/files?file-filters%5B%5D=.py&file-filters%5B%5D=.tflite#diff-07134a3031c180fdb9dc347a0d69837c0909fefddd733199210e3d34ae56db0fR74)

**Comment by jackwish** [comment]
TODO of TransposeConv (due to we do not fully understand the semantic gap) https://github.com/jackwish/tflite2onnx/pull/21#issuecomment-731760141
1. Special output shape for VALID padding
2. Different input/output shape for SAME padding

**Comment by jackwish** [comment]
changed the title to track the TransposeConv issue, will not track others anymore.

**Comment by jackwish** [comment]
all done

**Comment by ed-mr** [comment]
Is there any follow-up on this issue? Since I am not happy about the current decision, that onnx is not supporting NHWC order for half its operators. For all the folks using Conv and TF/Keras it would be the strait thing to support NHWC as onnx attribute in storage_order in the first right in the first place

**Comment by jackwish** [comment]
@ed-mr I am sorry but what's the contex are you refering to?

**Comment by ed-mr** [comment]
just in sort summary. I just stumbled over this old already closed issue. Which I would like to have questions about. Since to my understanding, it would be much better to improve onnx and provide the "storage_order" attribute to conv and gain abstraction. That way, tf2onnx can easily use its default data_format = "columns_last".
see [details on the issue I raised ealier](https://github.com/ed-mr/onnx-issue); also [onnx/tensorflow #1910](https://github.com/onnx/tensorflow-onnx/issues/1910) and [onnx #4099](https://github.com/onnx/onnx/discussions/4099)
I would be happy to see this, since it resolves issues on the backend sides. (I working with [tvm](https://github.com/apache/tvm/))

**Comment by ed-mr** [comment]
Besides the fact, that I would be in favour or the `storage_order`, I figure from the discussion I raised in the ONNX community [#4099](https://github.com/onnx/onnx/discussions/4099), the onnx-people might pick up such issue as local functions as described in [#3877](https://github.com/onnx/onnx/issues/3877). Reason sounds ok, they claim that ONNX operator basis becomes to big, since there are many operators to be adopted. I as a small user, and not a contributor I am certainly not able to convince anybody on that end. It might need some involvement from this project. Only issue I see is, to get the named operators NHWCConv on the list mentioned.

**Comment by jackwish** [comment]
@ed-mr Thanks, let me commont on the thread you initialized.
